### PR TITLE
Pr5

### PR DESCRIPTION
**Pull request by meiirbek-islamov**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.pytest_cache
+defects.egg-info/PKG-INFO
```
Comment (paulboone) [inline .gitignore]: *.egg-info

File: defects.egg-info/SOURCES.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+README.md
+setup.py
+defects.egg-info/PKG-INFO
+defects.egg-info/SOURCES.txt
+defects.egg-info/dependency_links.txt
+defects.egg-info/requires.txt
+defects.egg-info/top_level.txt
```
Comment (paulboone) [inline defects.egg-info/SOURCES.txt]: none of this should be added–this should be removed

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+.pytest_cache
+*.egg-info
```
Comment (paulboone) [inline .gitignore:2]: you need __pycache__ in here too. You accidentally committed a __pycache__file in the prior commit.

**Review by paulboone — CHANGES_REQUESTED** [review]
This is looking SO much better!! The only thing I see that still needs to be fixed is that you accidentally committed a __pycache__ file that should be removed. See my other comment.

**Review by paulboone — APPROVED** [review]
You got some great practice rebasing with this PR! It looks good. At some point, if it makes sense, you should go back and clean up Introduce_Defects.py (filename should definitely not have caps, for starters!), but I know we were just trying to get it into git at this point. I'm going to merge it.

**Comment by paulboone** [comment]
Ok, so one more problem! You actually rebased back into the master commit history (which you should never do), so now we have a merge conflict merging into master. We can manually resolve the conflicts and then merge. It will be a little ugly, but it is OK for your first PR. Just make sure that you don't rebase too far back next time. Probably easiest to fix it up over a zoom call so we can move on to the next PR...? I want you to continue doing new work rather than working on this one too much more.

**Comment by meiirbek-islamov** [comment]
Hi paul, 

Yeah, this was a great rebasing practice for me. I can meet today over zoom to quickly resolve conflicts and continue working further. I already finished working this defect project but I haven't PR'd yet. They are in another branch. I would like to merge them and start search and replace project immediately. I already created a repo and setup.py for that project. I called it functionaliseMOF. If we meet today I would like to discuss also the details of that project too.

**Pull request merged**